### ISSUE TITLE
humanizeDuration is not defined

### ISSUE THREAD

**Issue by jeffryang24** [siddii/angular-timer#214]
I use requirejs for loading my module. In requirejs, I add below code:

```
require.config({
          baseUrl: 'libs'.
          paths: {
                 'angular': 'path/to/angular',
                 'angular-timer': 'angular-timer/dist/angular-timer.min',
                 'humanize-duration': 'path/to/humanize',
                 'moment': 'path/to/moment'
          },
         shim: {
                'angular-timer': ['angular', 'moment', 'humanize-duration']
         }
});
```

But, the console always said that humanizeDuration is not defined, but moment is defined inside angular-timer.min.js

**Comment by jeffryang24** [comment]
Solved!!!! I just add these code to the beginning line of angular-timer.min.js

```

var humanizeDuration = require("humanize-duration");   // same like my requirejs shim. Also from the humanizeDuration docs at https://github.com/EvanHahn/HumanizeDuration.js
var moment = require("moment");  // same like my requirejs shim

**Comment by shonatodc** [comment]
This did NOT work, and the fact that we have to add this to core code smells. This shouldn't be closed, it's still an issue.

**Comment by siddii** [comment]
agreed.. been wanting to refactor/remove this dependency. But, haven't got a chance

**Comment by skotturi** [comment]
any update on this? Looks like these dependencies are still there and the lib doesnt work with the way its currently documented (just importing `lib/angular-timer/dist/angular-timer.js` )

**Comment by PradipShrestha** [comment]
Any update on this. I am facing same error with all js file reference.

**Comment by PradipShrestha** [comment]
The way I am currently able to work with webpack is to add plugin as:
`new webpack.ProvidePlugin({
            'moment': 'moment',
            'humanizeDuration': 'humanize-duration'
        })`

**Comment by jacqueslareau** [comment]
This issue needs to be reopened.

**Comment by Leo-G** [comment]
yup just faced this issue

**Comment by Leo-G** [comment]
not sure how i should fix as i do not use require or webpack

**Comment by PradipShrestha** [comment]
Leo-G: In that case, won't adding dependencies js: moment and humanizeDuration before angular-timer fix the issue?

**Comment by Leo-G** [comment]
I was hoping I did not have to install and load more dependencies in my app, but if thats the best solution then will do it

**Comment by mattisx** [comment]
I'm just writing my own timer, I can't stand any unnecessary dependencies.

**Comment by m477r33d** [comment]
+1 same issue.

**Comment by pint12** [comment]
having same issue 
why this issue is closed ?

**Comment by lfmedina** [comment]
+1 same issue.

**Comment by ybaruchel** [comment]
Same issue here..

**Comment by evankleist** [comment]
Same issue

**Comment by johntiger1** [comment]
same

**Comment by jmooreweb** [comment]
I got it to work by adding humanize-duration.js and moment.js after angular-timer.min.js.  
Its dependent on these other 2 libraries.

<script src="../lib/angular-1.6.6/angular.js"></script>

<script src="js/angular-timer.min.js"></script>
<script src="js/humanize-duration.js"></script>
<script src="js/moment.js"></script>

Get this libraries from Github.

**Comment by jijeshkj** [comment]
i have the same issue  "humanizeDuration is not defined" in including angular timer   
how to solve that ?

**Comment by kondalraodurgam** [comment]
I'm getting same error humanizeDuration  is not defined...???
